thanosql/resources/_record.py

```diff
@@ -0,0 +1,22 @@
+from typing import Any, List, Optional
+
+import pandas as pd
+from pydantic import BaseModel, model_validator
+
+
+class Record(BaseModel):
+    records: Optional[List[dict]] = None
+    total: Optional[int] = None
```

Comment:
```suggestion
    records: Optional[List[dict]]
    total: int
```
Do we need the default values? If I request a query like '~ limit 0' to our Engine API, what value comes into payload?

Reply:
I think default values are still needed, but instead of None I can set it to empty list and 0. I guess it should make more sense (?)

So that when ~ limit 0 is requested, the result will be something like Records(records=[], total=0)

Reply:
1. Records(records=None, total=None) or just None
2. Records(records=[], total=0)
3. Records(records=[v], total=l)

These are different, please check the cases!